### PR TITLE
Revert newcryo

### PR DESCRIPTION
**Pull request by SonixApache**
Reverts #24475.
Discussion with medbay players brought to my attention that these changes only really made the experience worse for everyone, medbay or otherwise.
Agree? Disagree? Sound off in the comments below! Be sure to subscribe and leave a 👍.

🆑 
 - rscdel: Recent cryo changes have been reverted, and cryo chem duping is back.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
Don't really know what to say, on one hand cryo had to be changed but on the other hand I don't know how do to a cryomix now

**Comment by MadmanMartian** [comment]
Its the same mix, just you need to make more of it

**Comment by Accipitermaris** [comment]
I don't think it needs reverting outright, but from what I've seen, I think it does need some tweaking. There's no emoji for that, though.

Even if you use the New Cryo system “properly<sup>[1]</sup>”, there are still many situations where a pile of guys need to be cloned, either because they were deeply husked by burns, or they were exposed to space for too long and have 855 brute. In those situations, New Cryo seems to drain quite quickly, even if you micromanage the injections and use a mix that is purely Clonex/Cryox/Alk, with Rye pills nearby. That seems like a meaningless repetition and micromanagement which doesn't actually punish lazy doctoring, as the recommended treatment method is still the same. (The prescribed treatment is still cloning.) If the goal is to make speedcloning less desirable regardless, we could always add an additional malus incurred by ejecting patients from cloning tubes prematurely.

<sup>[1]</sup>Where using cryo "properly" is defined as only using cryo for cloned patients whose bodies were too mangled to be recovered with surgery in a fashion that doesn't turn them into a nugget or chop off their head.

**Comment by brndd** [comment]
I'd like to see cryo cells have an internal beaker of like 500u or something so you don't have to sacrifice your own beakers to the tubes and can just fill it up when you have the time. Busywork or not, this change has led to the tubes being used less as a cure-all, which is good in my books.

--------------

On the topic of cloning, I don't think the fix to that lies in changing how cryo tubes work. I'm not a big fan of cloning as a mechanic purely because of RP reasons. It makes it too easy to bring characters that really should be permanently dead back into the round with all their knowledge. This results in a lot of metagaming-type issues when it comes to the topic of what led to the character's death.

I think it'd be better to severely limit the ability to clone dead characters by removing the ability to scan dead bodies with the cloning scanner. This way cloning backups would become more important; cloning memory loss is a bit easier to understand with those, too. To make up for this loss, perhaps something could be added to allow dead players to be brought back into the round as a randomname randombody that knows nothing about the player's past life -- effectively respawns, but with certain limitations such as requiring the dead player's old body to be recovered before it's possible.

**Comment by Tokanova** [comment]
@brndd  Internal beakers for the toobs would be amazing. Just make sure it has a ID lock so just anyone can't dump bad chems into it and force you to purge ALL the other chems

as for the topic at hand
I liked old Cryo because it effectively made life easier as a non medbay man, sometimes you just want to save someones life and theres no medical staff. Old style tube let you just put the default cryo juice that spawns near cryo in and not have to worry about it running out in 5 seconds.

**Comment by Blithering** [comment]
I can see the arguments for both sides of this one, cryo duping was useful for knowing you had a good mix waiting for those who need it, but may cause complacency where no-one checks to notice that someone has slipped bone hurting juice in there because the mix is "done". If the cryo changes were intended as a cloning nerf however, I think it may have missed the mark somewhat. I think the best way to make cloning less of the default option isn't mechanical nerfs, but convincing players their original body is valuable enough to hold on to. Nexis' proposed mechanical limb upgrades are promising on that front.

**Comment by Kammerjunk** [comment]
to reiterate what I misguidedly thought to say in the D*scord:
- this rework happened for reasons (that it accomplished):
  1. preventing reagents that are intentionally rare from being duplicated in volune by exploiting how cryo formerly worked
  2. making cryo a less attractive choice compared to the end-all solution that could be applied to every patient because the mix was nigh-infinite
- the above reasons are both good goals
- a proposed alternative to this implementation towards accomplish goal 2 was (by suggestion) to make cryox and clonex only heal clone damage. a such implementation would take away a feature of medbay. this one *does not,* but instead punishes repeated use of it.
- another proposed alternative for purposes of goal 2 was making non-cryo methods of returning a character (not a player) to full health (i.e. pills and surgery) better instead of making cryo worse. I can't see any ways in which pills and surgery aren't "literally fine," but many ways in which cryo was not "literally fine" in that it was too powerful.
- having a bad time in medbay is not a function of how cryo (currently) works, but a function of medbay in general. if medbay is staffed by bad players (where "bad" is usually shorthand for "inexperienced"), you do not and should not get treated nearly as well as if medbay is staffed by good players. this is *good*. you're hurt or dead, that's what happens. medbay is a messy experience on both the giving and receiving ends. in this regard, it's the same as all of the other departments: if there's nobody home, it's DIY day, so to speak.

the above points have importance in ascending order

beyond that, there are aspects of the rework that were good and unrelated to the current discussion:
- cryo no longer passively healing
- cryo consumes power
- upgrading cryo does literally anything

@Accipitermaris speedclones eject prematurely necessarily because that's what speedcloning is

**Comment by jknpj** [comment]
>free dex+, bicard, derma and antitox
>passive healing that needs no micromanaging at all
>consumes no power whatsoever

No, i'm not talking about old cryo. I'm talking about advanced medibots.

**Comment by Kammerjunk** [comment]
image
same

**Comment by aceedex** [comment]
yeah Im down. Never liked the solution to a problem being "make it a pain in the ass to do"
specially when it requires doing the shit that you do rounstart before you can have fun.
It's like saying that the problem is that people can print too much shit so we need to make it harder by having research reset after printing a few things. you should definitely keep a few of the features though, like the mechanics upgrades rather than a full on revert. 

Full disclosure. Cyro and cloning being a golden solution is fine by me if this is the alternative.

**Comment by Kammerjunk** [comment]
I'm not sure that that comparison works

**Comment by ShiftyRail** [comment]
I'm convinced the only people who actually support this change are dead inside and want the other players to feel their hollowness by forcing them to slave away at the chem dispenser

**Comment by Kammerjunk** [comment]
this is also how I feel about this PR
thank you for your support, Shifty

**Comment by PrimeDSS13** [comment]
Personally, I think nucryo is a chore. I don't really see the need to skillcap medbay further when we have a slight shortage on doctors anyways. The change hasn't really done anything RP-wise either, and I feel like I actually spend more time in the tube after I get cloned. People still just clone with 20 brute anyways.

File: __DEFINES/__compile_options.dm [hunk 1]
```diff
@@ -36,18 +36,18 @@
 
 // Toggles several features, explained in their respective comments.
 // You can turn those on and off manually if you prefer, instead of setting this
-#define DEVELOPER_MODE 0
+#define DEVELOPER_MODE 1
```
Comment (DamianX) [inline __DEFINES/__compile_options.dm]: delete this

Comment (SonixApache) [reply __DEFINES/__compile_options.dm]: goddamnit, i was wondering where my compile options went

File: code/modules/reagents/Chemistry-Holder.dm [hunk 2]
```diff
@@ -551,7 +551,7 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 		my_atom.on_reagent_change()
 	return 0
 
-/datum/reagents/proc/reaction(var/atom/A, var/method=TOUCH, var/volume_modifier=0, var/remove_reagents = FALSE)
```
Comment (jknpj) [inline code/modules/reagents/Chemistry-Holder.dm:554]: Why don't you change just cryo code's remove_reagents to FALSE instead

**Comment by juanototo2** [comment]
Some concessions must be made to make Medical playable and not waste the time of all parties involved. Until conversion gamemodes, blob, xenos, and mass casualty events in general are removed, there will be a need for cryo and cloning because they simply do not function without them - they are easy-to-use, fast, and provide the best method of shoving patients without broken bones out the door. Oldcryo IS one of these concessions - plus, nobody sane wants to add plasma to a breaker and press 14 buttons every 5 minutes because of how fast cryo drains.

**Comment by MadmanMartian** [comment]
All it is is now you've gotta make more mix than the starting mix, and properly strip your patients for it to do anything.

Before if you didn't strip your patients, it would eat the mix entirely.

**Comment by Kammerjunk** [comment]
what @MadmanMartian said
"medbay is unplayable" is hysteria

**Comment by brndd** [comment]
This server lives and diies by kneejerk reactions.

**Comment by ShiftyRail** [comment]
The point is less than medbay is unplayable and more than, from the little ventures I had in medbay recently, it's even more tedious and repetitive than before.

I'm not even sure what was the initial reasoning for it besides "tubes bad".
@MadmanMartian I think the rate at which cryomix was consumed by treating a clothed patient was enough before, and if that's the main issue this reworks tried to address, then by all means make it higher

**Comment by Kammerjunk** [comment]
it isn't.

**Comment by ShiftyRail** [comment]
https://www.youtube.com/watch?v=qVvZdfZSTCs

**Comment by Kammerjunk** [comment]
the repetitiveness isn't what isn't

**Comment by MadmanMartian** [comment]
Shifty, for the 5th time this thread, the changes were to remove chemical duping, stop acting dumb.

**Comment by ShiftyRail** [comment]
I see it now
I don't feel the need for such a change at all, I personally did not feel the need to explicitly use duping when I played medbay regularly. Maybe it's a much bigger problem now, but to me it wasn't a bad thing

besides cryoduping is such an ugly word, I'd rather call it "enhanced reagent distribution in refrigerated environment"

File: code/modules/reagents/Chemistry-Holder.dm (hunk 2 above)
Comment (SonixApache) [reply code/modules/reagents/Chemistry-Holder.dm:554]: Wouldn't that completely disable the depletion of the mix? That's not really the intention either.

**Comment by Kammerjunk** [comment]
"duping" is an exact description of the function and result of the entire thing
it is how cryo, prior to the change that this PR would revert, functioned and it is the functionality that this PR seeks to return to
duping is at the core of the former functionality of cryo and whether explicit or implicit, you *always* used it for cryo

**Comment by PrimeDSS13** [comment]
The emojiocracy has spoken, long live the emojiocracy.

File: code/modules/reagents/Chemistry-Holder.dm (hunk 2 above)
Comment (jknpj) [reply code/modules/reagents/Chemistry-Holder.dm:554]: No?

**Comment by brndd** [comment]
"I don't play medbay, BUT I think it's tedious"

The change accomplished what it set out to do: make tubes less of a cure-all solution (at least IMO that's what it set out to do; @MadmanMartian puts it as "removing reagent duping", but that's just hedging if you ask me). Far fewer people are stuffed in tubes for injuries that don't need the tubes now, because the doctors are punished for overusing cryo by having to constantly make new mix.

Because the rework/nerf PR also included changes besides "cryo uses more juice now", reverting it wholesale like this is stupid anyway. You should just readd reagent duping if you want the tubes to heal them with no effort, and keep in the parts upgrades and hardsuits no longer sucking out all the juice.

------

We should probably just replace medbay with a conveyor belt apparatus where dead corpses go in one end and living humans come out the other end anyway.

**Comment by ShiftyRail** [comment]
I played medbay almost continuously for a year doing a variety of things (chems, cryo, virus curing, surgery) depending on the situation before stopping almost entirely because of how dull and boring the time spent at the chem dispenser was.

This change - from little I played medical again - makes the time spent at the chem dispenser a much bigger part of playing medical, and something you have to do (because else the mix is running out) instead of something you choose to do (to make a good fridge).

**Comment by brndd** [comment]
You don't need to slave at the chem dispenser unless you use cryo for patients that don't actually need it. If you do find yourself constantly making more mix, that's an indication that you're overusing the tubes, and there's a simple solution to that: don't use the tubes and use the pills and the sleepers instead.

**Comment by d3athrow** [comment]
make the chems last longer but keep the duping thing fixed. make cryo less effective on severe physical wounds encouraging surgery to fix it.

**Comment by SonixApache** [comment]
having to change the metabolism of chems without duping would be an interesting hassle, but it does seem like that was the original intent except shitcoded
severe wounds like what? we only have surgery for broken bones and organ damage and cryo heals neither of those

**Comment by MadmanMartian** [comment]
How could we make the chems last longer?

Reduced injection rate? Or lower the metabolism on cryox/clonex?

**Comment by MadmanMartian** [comment]
I think I know what can help, hang on

**Comment by MadmanMartian** [comment]
#25339 should remedy most complaints

**Comment by ShiftyRail** [comment]
What kind of physical damage that would require surgery to fix would also be fixed by cryo ?
Cryo can't heal broken bones or internal bleeding, unless of course you get Perixadon.
Conversely, I don't think there is a surgery that specifically deals with severe trauma, except maybe removing a limb and re-attaching a fresh one.

**Comment by PJB3005** [comment]
I've played medbay a fair amount and I don't think the change is *that* bad.

I absolutely do shy away from using the tubes in most circumstances, but I think that's fine. Keeping somebody alive "properly" is still fairly easy. 

Having to micro manage cryo mix really is not that bad if you get large beakers. Especially if you get bluespace/large bluespace beakers from science (the former of which really isn't that hard to make for them provided R&D is being done at all, it requires no special minerals). So if we revert this change, medbay loses another "upgrade" from R&D. 

Interesting side thought: both surgery and cloning are made significantly easier from upgrades by R&D now, whereas previously you could trivially run the tubes without them. Doing large amounts of surgery is a massive pain unless you get upgraded tools as always. 

Slaving away at the chem dispenser really isn't that bad either. You're forced to do it if you want to run medbay efficiently either way and this barely makes a difference in that.

Generally speaking these kinds of balance changes are better accepted when the alternative is buffed, but I don't see any way to do that here without making pills/surgery just waaaaaay too strong.

I do think the change to make reagent use lower would be good to try out before merging this. Another idea is to make cryox/clonex metabolise slower at the cost of less effectiveness against other damage types.

**Comment by brndd** [comment]
What if body temperature reduced the rate at which things are metabolized?

**Comment by SonixApache** [comment]
with the changes cryo is also unfun to use for *everyone else* including bar / sec meme cryo.

**Comment by Zth--** [comment]
#25339

**Comment by SonixApache** [comment]
yeah that doesn't actually address the issues though

**Pull request closed without merging**